src/ar_http_iface_server.erl

```diff
@@ -65,7 +69,9 @@ split_path(Path) ->
 do_start(Port, GatewayOpts) ->
 	Dispatch = cowboy_router:compile(?HTTP_IFACE_ROUTES),
 	HttpIfaceEnv = #{ dispatch => Dispatch },
-	ok = ar_semaphore:start_link(hash_list_semaphore, 1),
+	ok = ar_semaphore:start_link(hash_list_semaphore, ?MAX_PARALLEL_HASH_LIST_REQUESTS),
+	ok = ar_semaphore:start_link(arql_semaphore, ?MAX_PARALLEL_ARQL_REQUESTS),
```

Comment:
A small wish, could we assign the semaphore atoms to their corresponding HTTP environments here? Then we do not have to match the `gateway` key in `with_arql_semaphore_req_field`, what is basically resolving the type of server again inside the request handler. Also, there would be no concern that atoms have to match here and there.

Reply:
Done in 9026590